enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -104,11 +104,16 @@ private static String getAPIKeyFromRequest(RequestContext requestContext) {
                     // key must exist in specified location
                     if (APIConstants.SWAGGER_API_KEY_IN_HEADER.equalsIgnoreCase(securitySchemaDefinition.getIn())) {
                         if (requestContext.getHeaders().containsKey(securitySchemaDefinition.getName())) {
+                            // Remove the apiKey header being forwarded to the backend.
```

Comment:
for authorizion header removal, we do it in this level. not in auth filter level. I'm wondering whether this should also be removed in this level

Reply:
https://github.com/wso2/product-microgateway/blob/1e53dfa16ee0ecce7c43594927e5d1444772dc5c/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java#L163

Reply:
Doing it outside auth filter is not practical in this case, because the header name is resolved in this level.

Reply:
If that is the case, whole process needs to be restructured rather than fixing the bug.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2563/commits/9ca6828292e1cc968c8af8031e58816a1b85a49a